### ISSUE TITLE
Mailchimp & Elementor Form: Let visitors choose categories/groups in one form

### ISSUE THREAD

**Issue by Maenggu** [elementor/elementor#7723]
## Prerequisites

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
Visitors should be able to choose in the Elementor form for which category they want to get our Newsletter. Since I have several categories and many different combinations are possible, I have to create an Elementor form (which is allocated to the according Mailchimp group in the settings) for every single category. I find it too complicated for visitors, if they have to put in their email address in several different Elementor forms to opt-in for more than one category.

**Describe the solution you'd like**
It would be great if visitors would be able to select the Newsletter categories/interests (which are the same as the groups in Mailchimp) they would like to opt-in for in checkbox fields. The data should be "delivered" via the API to Mailchimp, so that Mailchimp knows that if one of the checkboxes is checked, the visitor wants to sign up for this Mailchimp group (like it is today if you allocate a Mailchimp group yourself to an Elementor form).

**Describe alternatives you've considered**
The selection of the groups/categories could also be like tags (so that visitors can simply click on it), it doesn't have to be checkboxes necessarily (though I personally would prefer them).

Thank you very much in advance.

**Comment by get8bit** [comment]
I could also use this feature!

**Comment by SimonKarno** [comment]
Yeah, this is super useful. If I'd known Elementor didn't support it, I wouldn't have built the form using it.

**Comment by gmessieractivis** [comment]
I would also like this feature to be implemented!

**Comment by bodhi1970** [comment]
Super important! Please implement this soon.

**Comment by AndersJpg** [comment]
100% in for this. not sure why we cant already

**Comment by henningjp** [comment]
MailChimp has gone to one audience with multiple groups to segregate between interests.  These group choices (like Interest) show up on the MailChimp signup form, but I can't get that to work in Elementor (different problem).
While the Elementor form can be told to apply one or more groups, this cannot be linked to a set of checkboxes in the form.  With the changes in MailChimp structure, this becomes a necessary feature.

Please add a link between a multiple-item checkbox field and the MailChimp group choices from the audience selected.

**Comment by ashkan-ahmadi** [comment]
I totally agree. this is a must-have feature. The user must be able to pick what group they want to join

**Comment by henningjp** [comment]
So, I have a workaround, which is to build the form using the free plug-in, **MC4WP: Mailchimp for WordPress**.  It's not as "pretty" or as flexible as the Elementor forms, but it does give you user selectable check boxes for each of your Interest groups.  If a free plugin can do this, my **_paid_** subscription to Elementor Pro should be able to do it.  Let's go Elementor.

**Comment by DanielsRogulis** [comment]
We need this feature. It's super important!

**Comment by itismebrian** [comment]
I really don't understand why this feature hasn't been developed yet.

**Comment by GabeOddball** [comment]
I am just here to add my name to this request. My agency has been a pro user for several years and this feature is clearly a must.

**Comment by wimjuste** [comment]
+1 for me. 
Feature request to have the user be able to select the group he/she wants to join.

**Comment by Rocco-Contillo** [comment]
+1

The lack of this feature is the **only reason** I use Gravity Forms (or exceptionally CF7) on my own and my clients' websites. I do not understand why it has not been developed ... I think it is fairly trivial work to do.
I hope it will be introduced soon so I can start using just the Elementor form to be able to do **everything**!

Please Elementor team, give us an **indicative date** when this feature will be introduced! It's really **too important**.

**Comment by marcelk1101** [comment]
+1 exactly what I need too for Elementor!

**Comment by devinriggins** [comment]
I've been waiting on this feature for almost 3 years now. It's quite disappointing as it's pretty critical for Mailchimp integrations.

**Comment by itismebrian** [comment]
+1 Please add this feature!

**Comment by sittingpilgrim** [comment]
+1 for me as well

**Comment by edward-adams** [comment]
+1 — I'm astounded that this doesn't work. It was the last step in building out my form and totally killed my momentum. Please add this.

**Comment by contemplate** [comment]
+1 - please add

**Comment by acal** [comment]
+1 this would really get me using the forms a lot more and other 3rd party plugins less. Thanks!

**Comment by w3b-beweb** [comment]
+1 - please add

**Comment by Thinkmonger** [comment]
+1 Definitely add selectable Groups to this form!

**Comment by danny-joe** [comment]
As an Elementor Pro subscriber of several years, I'm REALLY at a loss as to why Elementor still hasn't integrated fully with MailChimp. Allowing subscribers to self-select Groups in a form, is pretty basic functionality provided by the competition (e.g. Divi). Instead I've had to hack things together with 3rd party form plugins that don't integrate well. I'm having to do any NEW websites in Divi, so I can avoid these Elementor limitations, despite preferring Elementor. (It's laughable that the Elementor site still says, "Note: For the moment, the visitors of the websites will not be able to self-select their group." Perhaps just remove the "For the moment" crap until you actually prioritize the work on that.)

**Comment by SimonKarno** [comment]
I use MC4WP: Mailchimp for WordPress and CF7 in Elementor @danny-joe It's a bit of a faff but it works well. Obv. much better if mailchimp groups were integrated into Elementor

**Comment by danny-joe** [comment]
Thanks @SimonKarno! I now have "MC4WP: MailChimp for Wordpress" Plugin installed/activated, but confused about "CF7 in Elementor"? I installed "Contact Form 7" plugin, assuming I need that too, and now struggling with which plugIn to install to make a "Contact Form 7" widget appear in Elementor. (I couldn't find one named "CF7 in Elementor".) Also confused about whether I create the form in MC4WP or in Contact Form 7? Any help MUCH appreciated!

**Comment by SimonKarno** [comment]
Right, yeah it's a bit counter intuitive.
- I use MC4WP for integration with Mailchimp.
- I use groups to segment my audience. A group sits in a category. Both Category and group have ID numbers
- I use CF7 for the form (not MC4WP)

1. connect MC4WP to Mailchimp using the API key in the usual way
2. once connected click on list name in MC4WP/Mailchimp page. You will see your groups below, as per screenshot
3. create a form in CF7
4. add MC4WP code as described in the docs: https://www.mc4wp.com/kb/integrations/
5. to use groups in the CF7 form (which I can't find in their docs, so no idea how I worked it out) see attached screenshots
6. Note: use the field names from the MC4WP Mailchimp settings page in the CF7 form
7. Note: use the category and group ids in the CF7 form as shown ie `name="mc4wp-INTERESTS[category name ID][]"` &  ` value="interests name ID"` _ensure you enter exactly as shown ie quoation marks_
8. Go to your Elementor page, click add widget. Search for form. You should see CF7 as an option. Add it to your page.
9. In the widget settings, choose your form from the content settings tab, style drop down.
10. You should now have a form that connects to Mailchimp groups
![MC4WP-settings](https://user-images.githubusercontent.com/55846844/226742343-3b644d11-3e41-4e39-a3fa-f0a3933730c1.png)
![CF7-code](https://user-images.githubusercontent.com/55846844/226742347-947faa04-a576-41ad-9e1b-053d1de8f3fe.png)
![Elementor widget](https://user-images.githubusercontent.com/55846844/226742453-3d95cf1d-f25c-43f8-b449-99c836e1dfdb.png)

**Comment by SimonKarno** [comment]
Hmm I just realised that you might need Elements Kit or some other plugin to display CF7 in Elementor. My instructions were based on a website I have just taken over, so the set up is a bit new to me. Other than the CF7 in Elementor issue, the instructions stand up.

**Comment by w3b-beweb** [comment]
> Hmm I just realised that you might need Elements Kit or some other plugin to display CF7 in Elementor. My instructions were based on a website I have just taken over, so the set up is a bit new to me. Other than the CF7 in Elementor issue, the instructions stand up.

CF7 has shortcodes to show forms, so you can  use a shortcode widget o directly a text editor widget and paste the form shortcode.

**Comment by SimonKarno** [comment]
> 

ah yeah, of course :-) Thanks for the tip @w3b-beweb

**Comment by danny-joe** [comment]
Alas, I was ultimately not able to get the prescribed combo to work. Never really got how the:
‹input name="mc4wp-INTERESTS[xxx][]" type="hidden" value="yyy" />
ultimately can be linked to an "Interests" [checkbox...] definition of a CF7 form so users can select their own interests and have that sent off to MailChimp with subscription request. The CF 7 spec for checkbox sections didn't *appear* to support mapping to MC groups or any other tag. (Perhaps there's a way, I just couldn't find it in documentation.)

But your efforts to document what you got to work for you, @SimonKarno,  is much appreciated... it got me further than I was.

Ultimately, I found a free plugin, Easy Forms for MailChimp (by YIKES), that supports user-specification of groups. Then just had to come up with the CSS.... much easier than the alternatives! (They DO provide an AddOn to the plugin that will do that, but I didn't want yet another subscription for something minor. So ultimately I was able to delete the MC4WP and CF7 and Elements Kit plugins in favor of Easy Forms for MailChimp.

**Comment by SimonKarno** [comment]
Ah, sorry I overlooked the bit about people choosing a group. Glad you got
it sorted

**Comment by BarBoost** [comment]
Three years later and.... nothing

**Comment by WebByCraig** [comment]
+1

**Comment by webbbio** [comment]
Just ran into this problem. Can't believe this problem is 5 years old and no results.

**Comment by devinriggins** [comment]
Welcome to the club

*Devin W. Riggins*
***@***.*** ***@***.***>*
*719-581-9326 | 443-797-9855*
Firefly List <https://fireflylist.com/>

**Comment by WebByCraig** [comment]
+1 There should be a way to assign groups based on checkboxes selected.

**Comment by hornsbycreative** [comment]
+1 still. 5 years after OP. Elementor please fix this! There should be a way to assign to Mailchimp groups based on checkboxes selected.

**Comment by BVLD** [comment]
+1 still. Six years later and no changes. Please Elementor fix this.
